src/lib/redux/userData.ts

```diff
@@ -20,13 +20,39 @@ const refreshUserData = async (user: any): Promise<void> => {
   }
 };
 
+//Function is called at a set interval and repulls data from backend 
+// set interval is 15 min/900000 ms
+const refreshDataBackground = async (): Promise<void> => {
+  try {
+    const state = store.getState();
+    //need to refresh data in the back ground
+    if (state.userData.isOnline) {
+      //refresh data code here
+      refreshSiteDataBackground();
+    }
+  } catch (err) {
+    console.log('Error occurred pulling data', err);
+  }
+}
+
 // Function is called at a set interval and updates redux's
 // isOnline value depending on if it gets a response from airlock
 const checkOnline = (): void => {
   $.ajax({
     url: process.env.REACT_APP_AIRTABLE_ENDPOINT_URL,
     type: 'GET',
-    success: () => store.dispatch(setIsOnline({ isOnline: true })),
+    success: () => {
+      //check if the prior state was offline, if state changes from offline -> online, reload data
+      const state = store.getState();
+      if (!state.userData.isOnline) {
+        //refresh data code here
+        store.dispatch(setIsOnline({ isOnline: true }))
+        refreshDataBackground();
+      }
+      store.dispatch(setIsOnline({ isOnline: true }))
```

Comment:
The `setIsOnline` doesn't have to be in the if statement since it'll be happening after it anyways right?